mirror-service/src/main/java/org/acme/client/NvdClient.java

```diff
@@ -0,0 +1,83 @@
+package org.acme.client;
+
+import io.github.jeremylong.nvdlib.NvdCveApi;
+import io.github.jeremylong.nvdlib.NvdCveApiBuilder;
+import io.github.jeremylong.nvdlib.nvd.DefCveItem;
+import org.apache.kafka.streams.processor.api.ContextualProcessor;
+import org.apache.kafka.streams.processor.api.ProcessorContext;
+import org.apache.kafka.streams.processor.api.Record;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.StoreBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.WebApplicationException;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.util.ArrayList;
+import java.util.Collection;
+
+/**
+ * Client for the NVD REST API.
+ */
+public class NvdClient extends ContextualProcessor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(NvdClient.class);
+    private static final String LAST_MODIFIED_EPOCH_KEY = "LAST_MODIFIED_EPOCH_KEY";
+    private KeyValueStore<String, Long> store;
+    private StoreBuilder<KeyValueStore<String, Long>> lastModifiedEpochStoreBuilder;
+    private String apiKey;
+
+    Collection<DefCveItem> nvdFeeds;
+
+    public NvdClient(StoreBuilder<KeyValueStore<String, Long>> lastModifiedEpochStoreBuilder,
+                     String apiKey) {
+        this.apiKey = apiKey;
+        this.nvdFeeds = new ArrayList<>();
+        this.lastModifiedEpochStoreBuilder = lastModifiedEpochStoreBuilder;
+    }
+
+    @Override
+    public void init(ProcessorContext context) {
+        super.init(context);
+        this.store = context().getStateStore(this.lastModifiedEpochStoreBuilder.name());
```

Comment:
The store has to be registered in the `Topology` somehow before it can be used. At the moment this will always return `null`.

Multiple ways to achieve this:

* Use a `ProcessorSupplier` and return the `StoreBuilder` in the [`stores`](https://kafka.apache.org/33/javadoc/org/apache/kafka/streams/processor/ConnectedStoreProvider.html#stores()) method. See `SnykProcessorSupplier` for reference: https://github.com/mehab/hyades/blob/68de966acea0fa88096e5852a700b722692b9548/vulnerability-analyzer/src/main/java/org/acme/processor/snyk/SnykProcessorSupplier.java#L59-L62
* Explicitly add the store to the `Topology` using [`streamsBuilder.addStateStore`](https://kafka.apache.org/0110/documentation/streams/developer-guide#streams_processor_statestore)

Once successfully registered, you should see a topic with a name along the lines of:

```
dtrack-mirror-service-nvd-last-modified-epoch-store-changelog
```

in Redpanda Console.